solutions/fully-configurable/main.tf

```diff
@@ -2,81 +2,59 @@
+  kms_region                = var.existing_kms_instance_crn != null ? module.existing_kms_crn_parser[0].region : var.existing_kms_root_key_crn != null ? module.existing_kms_key_crn_parser[0].region : null
+  existing_kms_guid         = var.existing_kms_instance_crn != null ? module.existing_kms_crn_parser[0].service_instance : var.existing_kms_root_key_crn != null ? module.existing_kms_key_crn_parser[0].service_instance : null
+  kms_service_name          = var.existing_kms_instance_crn != null ? module.existing_kms_crn_parser[0].service_name : var.existing_kms_root_key_crn != null ? module.existing_kms_key_crn_parser[0].service_name : null
+  kms_account_id            = var.existing_kms_instance_crn != null ? module.existing_kms_crn_parser[0].account_id : var.existing_kms_root_key_crn != null ? module.existing_kms_key_crn_parser[0].account_id : null
+  en_kms_key_id             = local.create_kms_keys ? module.kms[0].keys[format("%s.%s", local.en_key_ring_name, local.en_key_name)].key_id : var.existing_kms_root_key_crn != null ? module.existing_kms_key_crn_parser[0].resource : null
+  existing_kms_instance_crn = var.existing_kms_instance_crn != null ? var.existing_kms_instance_crn : "crn:v1:bluemix:${module.existing_kms_key_crn_parser[0].ctype}:${module.existing_kms_key_crn_parser[0].service_name}:${module.existing_kms_key_crn_parser[0].region}:${module.existing_kms_key_crn_parser[0].scope}:${module.existing_kms_key_crn_parser[0].service_instance}::"
+
+  # Create cross account EN / KMS auth policy if not using existing EN instance, if 'skip_en_kms_auth_policy' is false, and a value is passed for 'ibmcloud_key_management_service_api_key'
+  create_cross_account_en_kms_auth_policy = var.existing_event_notifications_instance_crn == null && !var.skip_event_notifications_kms_auth_policy && var.ibmcloud_kms_api_key != null
+  # Create cross account COS / KMS auth policy if not using existing EN instance, if 'skip_cos_kms_auth_policy' is false, and if a value is passed for 'ibmcloud_key_management_service_api_key'
+  create_cross_account_cos_kms_auth_policy = var.existing_event_notifications_instance_crn == null && !var.skip_cos_kms_auth_policy && var.ibmcloud_kms_api_key != null
   # If a prefix value is passed, add it to the EN key name
-  en_key_name = try("${local.prefix}-${var.en_key_name}", var.en_key_name)
+  en_key_name = try("${local.prefix}-${var.event_notifications_key_name}", var.event_notifications_key_name)
   # If a prefix value is passed, add it to the EN key ring name
-  en_key_ring_name = try("${local.prefix}-${var.en_key_ring_name}", var.en_key_ring_name)
-  # If a prefix value is passed, add it to the COS key name
+  en_key_ring_name = try("${local.prefix}-${var.event_notifications_key_ring_name}", var.event_notifications_key_ring_name)
+  # Use existing key if set. Else if new key and if a prefix value is passed, add it to the COS key name
   cos_key_name = try("${local.prefix}-${var.cos_key_name}", var.cos_key_name)
```

Comment:
if you add `-` to the local.prefix value
` prefix = var.prefix != null ? trimspace(var.prefix) != "" ? "${var.prefix}-" : "" : ""` 
then you do not need `try` here, instead could be only
`"${local.prefix}${var.cos_key_name}"`